Detectors/ITSMFT/common/reconstruction/src/Clusterer.cxx

```diff
@@ -172,106 +170,86 @@ void Clusterer::updateChip(UInt_t ip)
-    }
-    mPreClusterIndices[i1] = -1;
-    for (Int_t i2 = i1 + 1; i2 < mPreClusterHeads.size(); ++i2) {
-      if (mPreClusterIndices[i2] != ci) {
-        continue;
-      }
-      next = mPreClusterHeads[i2];
-      while (next >= 0) {
-        const auto& dig = mPixels[next];
-        const auto pix = pixData[dig.second]; // PixelData
-        x += pix.getRowDirect();
-        z += pix.getCol();
+        mPixArrBuff[npix++] = pix; // needed for cluster topology
         if (pix.getRowDirect() < rowMin) {
           rowMin = pix.getRowDirect();
-        }
-        if (pix.getRowDirect() > rowMax) {
+        } else if (pix.getRowDirect() > rowMax) {
           rowMax = pix.getRowDirect();
         }
         if (pix.getCol() < colMin) {
           colMin = pix.getCol();
-        }
-        if (pix.getCol() > colMax) {
+        } else if (pix.getCol() > colMax) {
           colMax = pix.getCol();
         }
-        if (npix < mPixArrBuff.size()) {
-          mPixArrBuff[npix] = pix; // needed for cluster topology
-        }
         if (labelsClus) { // the MCtruth for this pixel is at mChipData->startID+dig.second
           fetchMCLabels(dig.second + mChipData->getStartID(), labelsDig, nlab);
         }
-        npix++;
         next = dig.first;
+      } else {
+        LOG(ERROR) << "Cluster size " << npix + 1 << " exceeds the buffer size" << FairLogger::endl;
       }
-      mPreClusterIndices[i2] = -1;
     }
-
-    Point3D<float> xyzLoc(Segmentation::getFirstRowCoordinate() + x * Segmentation::PitchRow / npix, 0.f,
-                          Segmentation::getFirstColCoordinate() + z * Segmentation::PitchCol / npix);
-    auto xyzTra =
-      mGeometry->getMatrixT2L(mChipData->getChipID()) ^ (xyzLoc); // inverse transform from Local to Tracking frame
-
-    clusters.emplace_back();
-    Cluster& c = clusters[noc];
-    c.setROFrame(mChipData->getROFrame());
-    c.setSensorID(mChipData->getChipID());
-    c.setPos(xyzTra);
-    c.setErrors(SigmaX2, SigmaY2, 0.f);
-    c.setNxNzN(rowMax - rowMin + 1, colMax - colMin + 1, npix);
-    if (labelsClus) { // MC labels were requested
-      for (int i = nlab; i--;) {
-        labelsClus->addElement(noc, mLabelsBuff[i]);
+    mPreClusterIndices[i1] = -1;
+    for (Int_t i2 = i1 + 1; i2 < mPreClusterHeads.size(); ++i2) {
+      if (mPreClusterIndices[i2] != ci) {
+        continue;
+      }
+      next = mPreClusterHeads[i2];
+      while (next >= 0) {
+        const auto& dig = mPixels[next];
+        const auto pix = pixData[dig.second]; // PixelData
+        if (npix < mPixArrBuff.size()) {
+          mPixArrBuff[npix++] = pix; // needed for cluster topology
+          if (pix.getRowDirect() < rowMin) {
+            rowMin = pix.getRowDirect();
+          } else if (pix.getRowDirect() > rowMax) {
+            rowMax = pix.getRowDirect();
+          }
+          if (pix.getCol() < colMin) {
+            colMin = pix.getCol();
+          } else if (pix.getCol() > colMax) {
```

Comment:
The checks with respect to colMin and colMax must be independent. No "else", right ?